ui/ops/src/dynamic/widgets/graphic/widgets.js

```diff
@@ -25,48 +25,55 @@ export function useWidgetEffects(el, config, sources) {
 
   // allow selecting a child element for the widget to replace
   if (effect.selector) {
-    el = el.querySelector(effect.selector);
+    const els = el.querySelectorAll(effect.selector);
     if (!el) {
       console.warn(`Could not find child element with selector: ${effect.selector}`);
       return null;
     }
-  }
 
-  const {rootWidth, rootHeight} = svgRootSize(el);
-  const bounds = elementBounds(el);
-  const percent = (v, t) => {
-    return (v / t * 100).toFixed(4) + '%';
-  };
-  const boundsPercent = {
-    top: percent(bounds.y, rootHeight),
-    left: percent(bounds.x, rootWidth),
-    width: percent(bounds.width, rootWidth),
-    height: percent(bounds.height, rootHeight)
-  };
+    const ret = [];
 
-  if (!effect.showElement) {
-    el.style.visibility = 'hidden';
-  }
+    els.forEach(el => {
+      const {rootWidth, rootHeight} = svgRootSize(el);
+      const bounds = elementBounds(el);
+      const percent = (v, t) => {
+        return (v / t * 100).toFixed(4) + '%';
+      };
+      const boundsPercent = {
+        top: percent(bounds.y, rootHeight),
+        left: percent(bounds.x, rootWidth),
+        width: percent(bounds.width, rootWidth),
+        height: percent(bounds.height, rootHeight)
+      };
 
-  const props = {};
-  for (const [k, v] of Object.entries(effect.props)) {
-    if (typeof v === 'object' && 'ref' in v) {
-      const source = sources[v.ref];
-      if (source) {
-        props[k] = computed(() => _get(toValue(source.value), v.property));
-      } else {
-        console.warn(`Unknown source: ${v.ref}`);
+      if (!effect.showElement) {
+        el.style.visibility = 'hidden';
       }
-    } else {
-      props[k] = v;
-    }
-  }
 
-  return {
-    component: markRaw(comp),
-    bounds: boundsPercent,
-    props: reactive(props)
-  };
+      const props = {};
+      for (const [k, v] of Object.entries(effect.props)) {
+        if (typeof v === 'object' && 'ref' in v) {
+          const source = sources[v.ref];
+          if (source) {
+            props[k] = computed(() => _get(toValue(source.value), v.property));
+          } else {
+            console.warn(`Unknown source: ${v.ref}`);
+          }
+        } else {
+          props[k] = v;
+        }
+      }
+
+      ret.push({
+        component: markRaw(comp),
+        bounds: boundsPercent,
+        props: reactive(props)
+      });
+
+    });
+    return ret;
+  }
+  return null;
```

Comment:
You've removed support for effects without a selector. Can you add it back in please.